### PR TITLE
[dv,chip,clkmgr] Add clkmgr_sleep_frequency test

### PR DESCRIPTION
**Pull request by matutem**
Signed-off-by: Guillermo Maturana <maturana@google.com>

### CODE REVIEW

File: sw/device/tests/clkmgr_sleep_frequency_test.c [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+static pwrmgr_isr_ctx_t pwrmgr_isr_ctx = {
+    .pwrmgr = &pwrmgr,
+    .plic_pwrmgr_start_irq_id = kTopEarlgreyPlicIrqIdPwrmgrAonWakeup,
+    .expected_irq = kDifPwrmgrIrqWakeup,
+    .is_only_irq = true};
+
+static volatile bool isr_entered;
+
+// Will flag a failure if any measurements are disabled.
+void check_all_measurements_are_enabled(const dif_clkmgr_t *clkmgr) {
```
Comment (jdonjdon) [inline sw/device/tests/clkmgr_sleep_frequency_test.c:61]: static void maybe?

**Review by jdonjdon — APPROVED** [review]
LGTM with a minor suggestion

File: sw/device/tests/clkmgr_sleep_frequency_test.c (hunk 1 above)
Comment (matutem) [reply sw/device/tests/clkmgr_sleep_frequency_test.c]: Done

File: sw/device/tests/clkmgr_sleep_frequency_test.c [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "sw/device/lib/testing/autogen/isr_testutils.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+/**
+ * This test measure clock counts with clkmgr.
+ * Frequency meaasurement assumes aon_clk frequency 200KHz.
```
[the remark this answers is not in the document]
Comment (matutem) [reply sw/device/tests/clkmgr_sleep_frequency_test.c:24]: I am creating a PR that makes the measurement counts determined by device.h, so this comment should be obsoleted.

[the remark this answers is not in the document]
Comment (matutem) [reply sw/device/tests/clkmgr_sleep_frequency_test.c:24]: See https://github.com/lowRISC/opentitan/pull/13592.

[the remark this answers is not in the document]
Comment (matutem) [reply sw/device/tests/clkmgr_sleep_frequency_test.c]: Fixed

**Review by sriyerg — APPROVED** [review]
LGTM.

File: sw/device/tests/clkmgr_sleep_frequency_test.c [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+ * After 100 measurements, test kicks in regular sleep with IO and USB
+ * clocks turned off. Once the chip wakes up the measurements should be
+ * enabled, and errors may be found for the disabled clocks.
+ *
+ * Notice the test overrides the hardware behavior so it comes out with
+ * calibrated USB clock, otherwise the USB clock frequency will be incorrect.
+ * USB calibration should be a separate test, and may be vendor-specific.
+ */
+enum {
+  kWaitForCSRPolling = 1,  // 1us
```
[the remark this answers is not in the document]
Comment (matutem) [reply sw/device/tests/clkmgr_sleep_frequency_test.c]: Done

**Pull request merged**